kubernetes/check.py

```diff
@@ -156,9 +176,15 @@ def check(self, instance):
         self._update_metrics(instance, pods_list)
 
         # kubelet events
-        if _is_affirmative(instance.get('collect_events', DEFAULT_COLLECT_EVENTS)):
+        if self.event_retriever is not None:
             try:
-                self._process_events(instance, pods_list)
+                events = self.event_retriever.get_event_array()
+                if events and self._collect_events:
+                    self._process_events(instance, pods_list, events)
+                changed_pods = self.kubeutil.process_events(events)
```

Comment:
renaming _process_events to _update_kube_events to match the other methods reporting data